### PR TITLE
Add objectSearch type

### PR DESCRIPTION
**Pull request by rkuykendall**
**JIRA:** https://thatsmighty.atlassian.net/browse/MTY-987

image
image

### CODE REVIEW

File: test/factories.ts [hunk 1]
```diff
@@ -86,7 +86,13 @@ export const radioFactory = fieldFactoryForType('radio')
 export const optionSelectFactory = fieldFactoryForType('optionSelect')
   .attrs({ options: attrOptions });
 
+export const objectSearchFactory = fieldFactoryForType('objectSearch')
+  .attrs({
+    endpoint: '/endpoint/',
+  });
+
 export const objectSearchCreateFactory = fieldFactoryForType('objectSearchCreate')
+  .extend(objectSearchFactory)
```
Comment (SHA-4) [inline test/factories.ts]: Wouldn't this extension override the type of `objectSearchCreateFactory`?

File: test/factories.ts [hunk 2]
```diff
@@ -86,7 +86,13 @@ export const radioFactory = fieldFactoryForType('radio')
...
   .attrs({
     createFields: [{ field: 'name', required: true }],
     endpoint: '/endpoint/',
```
Comment (SHA-4) [inline test/factories.ts]: If you do end up extending `objectSearchFactory`, this attribute is redundant

File: test/factories.ts (hunk 1 above)
Comment (rkuykendall) [reply test/factories.ts]: I suppose so?

File: test/factories.ts (hunk 2 above)
Comment (rkuykendall) [reply test/factories.ts]: I suppose so?

File: test/factories.ts (hunk 1 above)
Comment (rkuykendall) [reply test/factories.ts]: Seems fine

**Pull request merged**